src/main/java/net/atos/zac/app/informatieobjecten/InformatieObjectenRESTService.java

```diff
@@ -286,6 +287,13 @@ public RESTEnkelvoudigInformatieobject createEnkelvoudigInformatieobjectAndUploa
         final EnkelvoudigInformatieObjectData enkelvoudigInformatieObjectData = taakObject ?
                 informatieobjectConverter.convertTaakObject(restEnkelvoudigInformatieobject) :
                 informatieobjectConverter.convertZaakObject(restEnkelvoudigInformatieobject);
+
+        Integer fileSize = enkelvoudigInformatieObjectData.getBestandsomvang();
+        String fileName = enkelvoudigInformatieObjectData.getBestandsnaam();
+        if (fileName != null && (fileSize == null || fileSize.longValue() == 0)) {
```

Comment:
We should really use Jakarta validators for this I think? Those don't work yet in our Kotlin code but in our Java code they should be fine.

Reply:
Moved the logic to Jakarta ConstraintValidator